test/integration/DPOS.spec.js

```diff
@@ -0,0 +1,139 @@
+      await tdpos.initializeRound();
+      assert.deepEqual(roundNumber.add(1), await tdpos.roundNumber.call());
+    });
+
+    it('provider1 registers as Provider', async () => {
+      assert.equal(PROVIDER_UNREGISTERED, await tdpos.providerStatus(provider1));
+      await tdpos.provider(22, 10, 1, 25, {from: provider1});
+      assert.equal(PROVIDER_REGISTERED, await tdpos.providerStatus(provider1));
+    });
+
+    it('someone calls initializeRound again but it fails because election period of current round is not over', async () => {
+      await assertFail( tdpos.initializeRound() );
+    });
+
+    it("provider2 fails to register because he didn't delegate tokens to himself", async () => {
+      await assertFail( tdpos.provider(22, 10, 1, 25, {from: provider2}) );
+    });
+
+    it('provider2 delegates tokens to himself', async () => {
+      await tdpos.approve(contractAddress, 200, {from: provider2});
+      await tdpos.bond(provider2, 200, {from: provider2});
+    });
+
+    it('provider2 registers as Provider', async () => {
+      assert.equal(PROVIDER_UNREGISTERED, await tdpos.providerStatus(provider2));
+      await tdpos.provider(22, 10, 1, 25, {from: provider2});
+      assert.equal(PROVIDER_REGISTERED, await tdpos.providerStatus(provider2));
+    });
+
+    it('provider3 delegates tokens to himself', async () => {
+      await tdpos.approve(contractAddress, 300, {from: provider3});
+      await tdpos.bond(provider3, 300, {from: provider3});
+    });
+
+    it('provider3 fails to register with invalid parameters', async () => {
+      // Cannot have negative blockRewardCut
+      await assertFail( tdpos.provider(22, 10, -1, 25, {from: provider3}) );
+      // Cannot have over 100% of feeShare
+      await assertFail( tdpos.provider(22, 10, 1, 125, {from: provider3}) );
+    });
+
+    it('provider3 registers as Provider with valid parameters', async () => {
+      assert.equal(PROVIDER_UNREGISTERED, await tdpos.providerStatus(provider3));
+      await tdpos.provider(22, 10, 1, 25, {from: provider3});
+      assert.equal(PROVIDER_REGISTERED, await tdpos.providerStatus(provider3));
+    });
+
+    it('provider4 delegates tokens to himself', async () => {
+      await tdpos.approve(contractAddress, 400, {from: provider4});
+      await tdpos.bond(provider4, 400, {from: provider4});
+    });
+
+    it('provider4 fails to register because rateLockDeadline has passed for the current round', async () => {
+      const rateLockDeadlineBlock = await roundManagerHelper.getRateLockDeadlineBlock(tdpos);
+      await blockMiner.mineUntilBlock(rateLockDeadlineBlock);
+      await assertFail( tdpos.provider(22, 10, 1, 25, {from: provider4}) );
+    });
+
+    it('someone calls initializeRound() but it fails because Election Period is not over', async () => {
+      await assertFail( tdpos.initializeRound() );
+    });
+
+    it('someone calls initializeRound() after Election Period is over', async () => {
```

Comment:
I would call this test, initializeRound increments roundNumber after Election Period is over.

The text in the it, should describe expected behavior, not literal activity.